op-e2e/actions/ecotone_fork_test.go

```diff
@@ -132,14 +134,49 @@ func TestEcotoneNetworkUpgradeTransactions(gt *testing.T) {
 
 	// 4788 contract is deployed
 	expected4788Address := crypto.CreateAddress(derive.EIP4788From, 0)
-	code, err := engine.EthClient().CodeAt(context.Background(), expected4788Address, latestBlock.Number())
-	require.NoError(t, err)
-	require.Equal(t, code, predeploys.EIP4788ContractCode)
+	require.Equal(t, predeploys.EIP4788ContractAddr, expected4788Address)
+	code, err := ethCl.CodeAt(context.Background(), expected4788Address, latestBlock.Number())
+	require.NoError(t, err)
+	require.NotEmpty(t, code)
+	require.Equal(t, predeploys.EIP4788ContractCode, code)
+	require.NotEmpty(t, predeploys.EIP4788ContractAddr)
+	require.Equal(t, predeploys.EIP4788ContractCodeHash, crypto.Keccak256Hash(code))
+	// Test that the beacon-block-root has been set
+	checkBeaconBlockRoot := func(timestamp uint64, expectedHash common.Hash, expectedTime uint64, msg string) {
+		historyBufferLength := uint64(8191)
+		rootIdx := common.BigToHash(new(big.Int).SetUint64((timestamp % historyBufferLength) + historyBufferLength))
+		timeIdx := common.BigToHash(new(big.Int).SetUint64(timestamp % historyBufferLength))
+
+		rootValue, err := ethCl.StorageAt(context.Background(), predeploys.EIP4788ContractAddr, rootIdx, nil)
+		require.NoError(t, err)
+		require.Equal(t, expectedHash, common.BytesToHash(rootValue), msg)
+
+		timeValue, err := ethCl.StorageAt(context.Background(), predeploys.EIP4788ContractAddr, timeIdx, nil)
+		require.NoError(t, err)
+		require.True(t, common.BytesToHash(timeValue).Big().IsUint64())
+		require.Equal(t, expectedTime, common.BytesToHash(timeValue).Big().Uint64(), msg)
```

Comment:
why not
```suggestion
		timeBig := new(big.Int).SetBytes(timeValue)
		require.True(t, timeBig.IsUint64())
		require.Equal(t, expectedTime, timeBig.Uint64(), msg)
```
? Might be misleading to convert a number value to a "Hash" and then back to a `big.Int`.